### PR TITLE
Fix segault from bad GIL release

### PR DESCRIPTION
**Pull request by SimonHeybrock**
`py::cast` may not be used while GIL is released. This lead to a crash when turning on the profile view when plotting event data.

### CODE REVIEW

**Comment by jl-wynen** [comment]
I'm getting an infinite loop now! Does this require the zooming fix?

**Comment by SimonHeybrock** [comment]
In principle this should be independent, but it may well be that one issue hid the other.

**Comment by jl-wynen** [comment]
Are you on a clean branch for this?

**Comment by SimonHeybrock** [comment]
@jl-wynen Yes, works for me.

**Comment by jl-wynen** [comment]
I recompiled on a clean branch and still get the freeze. I'm using a debug build with multi-threading.

**Comment by SimonHeybrock** [comment]
@jl-wynen Does it also freeze if you apply your branch in addition?

**Comment by jl-wynen** [comment]
That did it, it's not freezing anymore.

**Review by jl-wynen — APPROVED** [review]
Approved, but only merge after #1583 has been merged.

**Pull request merged**
